llama-index-integrations/selectors/llama-index-selectors-notdiamond/tests/test_selectors_notdiamond.py

```diff
@@ -0,0 +1,95 @@
+import os
+import pytest
+from typing import List
+from unittest.mock import MagicMock, patch, AsyncMock
+import uuid
+
+from llama_index.core.base.base_selector import (
+    SelectorResult,
+    SingleSelection,
+)
+from llama_index.core.schema import QueryBundle
+from llama_index.core.tools import ToolMetadata
+from llama_index.selectors.notdiamond.base import NotDiamondSelector, LLMSingleSelector
+
+from notdiamond import LLMConfig
+
+
+@pytest.fixture()
+def session_id() -> str:
+    return str(uuid.uuid4())
+
+
+@pytest.fixture()
+def choices() -> List[ToolMetadata]:
+    return [
+        ToolMetadata(
+            name="vector_index", description="Great for asking questions about recipes."
+        ),
+        ToolMetadata(name="list_index", description="Great for summarizing recipes."),
+    ]
+
+
+@pytest.fixture()
+def nd_selector(session_id):
```

Comment:
We mock out the inherited attributes of a `NotDiamondSelector` here to focus on testing the class functionality. Mocks for ND API, as well as the parent class `_select` and `_aselect` to avoid unintended LLM client creations or calls.